arangod/Cluster/ClusterInfo.cpp

```diff
@@ -6600,53 +6670,87 @@ ClusterInfo::getCurrent(uint64_t& index,
 
 containers::FlatHashSet<ServerID> ClusterInfo::getFailedServers() const {
   std::lock_guard lock{_failedServersMutex};
-  return _failedServers;
+  containers::FlatHashSet<ServerID> result;
+  result.reserve(_failedServers.size());
+  for (auto const& server : _failedServers) {
+    result.emplace(server);
+  }
+  return result;
 }
 
 void ClusterInfo::setFailedServers(
     containers::FlatHashSet<ServerID> failedServers) {
   std::lock_guard lock{_failedServersMutex};
-  _failedServers = std::move(failedServers);
+  _failedServers.clear();
+  for (auto const& server : failedServers) {
+    _failedServers.emplace(server);
+  }
 }
 
 #ifdef ARANGODB_USE_GOOGLE_TESTS
 void ClusterInfo::setServers(
     containers::FlatHashMap<ServerID, std::string> servers) {
   WRITE_LOCKER(writeLocker, _serversProt.lock);
-  _servers = std::move(servers);
+  _servers.clear();
+  for (auto const& [k, v] : servers) {
+    _servers.emplace(k, v);
+  }
 }
 
 void ClusterInfo::setServerAliases(
     containers::FlatHashMap<ServerID, std::string> aliases) {
   WRITE_LOCKER(writeLocker, _serversProt.lock);
-  _serverAliases = std::move(aliases);
+  _serverAliases.clear();
+  for (auto const& [k, v] : aliases) {
+    _serverAliases.emplace(k, v);
```

Comment:
same (also for the following methods).

Reply:
only used when `ARANGODB_USE_GOOGLE_TESTS` is defined. So probably not even relevant for production.